onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -303,6 +314,40 @@ def eval_function(  # type: ignore[override]
         args: Sequence[ValidArgumentType],
         kwargs: Mapping[str, ValidArgumentType],
     ):
+        if _flags.EXPERIMENTAL_PREFER_TRACING:
+            # Special cases for handling IsScalar and Rank
+            if function.name == "IsScalar":
+                if len(args) != 1:
+                    raise TypeError(
+                        f"Expected 1 positional argument for function '{function}', got {len(args)}."
+                    )
+                if isinstance(args[0], TorchScriptTensor):
+                    if args[0].rank is not None:
+                        return args[0].rank == 0
+                    else:
+                        # Fall to call the function
+                        pass
+                else:
+                    # Python constants are scalars
+                    return True
+            if function.name == "Rank":
```

Comment:
Because rank is immutable, we can return int?

Reply:
Tracing is scattered in the codebase. I am a bit worried if it would become a debugging problem of dynamic/static for future?

Reply:
Do you mean the `traceable` flag? Only the functions that are safe to trace are marked as traceable. Do you foresee any issues?

Reply:
Yes. The assumption is that Rank is going to be static. This should be true even if we have dynamic axis. If that's not the case we can remove this experiment or find other ways around it (e.g. using the torch.export constraints).

Reply:
I guess it's the complexity of the flag bringing us. I think we are trying to keep "scripted as much as it can". However, the node modifications now could happen here, converter(filter_args_XXX), and torchlib. I wonder if we have a plan to make it simpler, so whenever we have an unexpected static axe model issue, we have clear targets where the code might be.

Reply:
Resolved offline. We agreed that this is an experimental feature that will be removed or adjusted in the future.